src/api/githubService.js

```diff
@@ -0,0 +1,284 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import axios from "axios";
+import moment from "moment";
+
+const functionalTestsWorkflowId = 5937682;
+
+export const getAllWorkflows = async () => {
+  const api = `https://api.github.com/repos/mozilla-mobile/mozilla-vpn-client/actions/runs?branch=main`;
+
+  let wfRunsInfo = [];
+  let workflows = {};
+
+  if (await _isWithinExpiryTime("awf_expireTime"))
+    return await _getDataFromLocal("all_workflows");
+
+  const {
+    data: { workflow_runs },
+  } = await axios.get(api);
+
+  for (let workflow_run of workflow_runs) {
+    if (workflows[workflow_run.name] === undefined) {
+      workflows[workflow_run.name] = workflow_run.id;
+      wfRunsInfo.push(workflow_run);
+    }
+  }
+
+  window.localStorage.setItem(
+    "awf_expireTime",
+    moment().add(45, "m").valueOf()
```

Comment:
in this way it's per client... why about if you do it in a github action any 30 minutes? To have fresh data each time and speed up the client rendering?

Reply:
its for rate limiting, only so many calls can be made per IP address within an hour window

Reply:
right... so, if we reach the limit, there are no waits we can do. The loading of the page cannot take more than 1 hour...